### PR TITLE
[ENH] Softmax activation function added

### PR DESCRIPTION
**Pull request by KowshikB03**
Fixes Issue #2593

### CODE REVIEW

**Comment by hadifawaz1999** [comment]
Thank you for this but i left a comment on the corresponding issue on how this is not a bug

**Comment by MatthewMiddlehurst** [comment]
Hi, I agree with Ali here if these are the published defaults then I don't really want to change them unless there is a publication showing it is better.

@hadifawaz1999 a warning or documentation change should be fine though if you think this would produce better results in certain scenarios? If not or it's unknown, keep it as it is. Up to you really.

**Comment by hadifawaz1999** [comment]
not sure how worth it it would be, its not really the best model at the end of the day, with or without softmax, so would just keep it as is and not add confusion to users

**Comment by MatthewMiddlehurst** [comment]
ok will leave it up for a bit for others to comment but likely declined

**Comment by KowshikB03** [comment]
In the comparison of multi-class datasets using the OsuLeaf dataset, the model with Softmax activation shows steady improvement in accuracy. Although it didn’t achieve the highest accuracy within 50 epochs, it may improve further with 100 epochs. On the other hand, the model with Sigmoid activation stabilizes around 0.67–0.68 after 40 epochs, with no consistent improvement in accuracy or loss reduction. This was the same pattern in basic_motions dataset training. The main point is that a simple CNN doesn't always give better accuracy without fine-tuning. Starting with better default functions makes fine-tuning easier, helping us reach 95+ accuracy instead of just 85+ for multi-class single-label datasets.

Adding articles from stanford, wandbai and few more that explain theoretical reasons why softmax is better form multi-class single-label datasets.

1. https://web.stanford.edu/~nanbhas/blog/sigmoid-softmax/
2. https://wandb.ai/ayush-thakur/dl-question-bank/reports/Comparing-Sigmoid-MSE-With-Softmax-Cross-Entropy-for-Image-Classification--VmlldzoyMDA3ODQ
3. https://glassboxmedicine.com/2019/05/26/classification-sigmoid-vs-softmax/

Maybe we can train for further more epochs like 200 for more convincing results. I am open for your suggestions.

**Comment by hadifawaz1999** [comment]
Just to be clear, every model in aeon uses softmax activation in the classification/deep_learning module, as we do in our research field. TimeCNN is an exception because this is how it was proposed in the original paper. It is not the best model as models such as FCN ResNet InceptionTime and LITETime are much more performant, so even configuring TimeCNN in a different way won't outperform the state-of-the-art. Moreover, this is not really the interest of aeon as the goal is to implement existing models that are proposed in the literature in a most optimized and organized way

What could be a possible additional information is a sentence in the TimeCNNClassifier documentation that mentions explicitly that this model uses sigmoid activation and mse loss function. Closing this PR and it's related issue for now, feel free to PR for the additional sentence in the docstring of the classifier

**Comment by KowshikB03** [comment]
Thank you for the clarification. 

I understand the Aeon interests, and appreciate your insights. I will create a new PR to update the TimeCNNClassifier documentation accordingly. We can close this PR
